### PR TITLE
Add usdt staking

### PR DESCRIPTION
**Pull request by jerry-yu**
Add usdt staking

### CODE REVIEW

File: pallets/credit/src/lib.rs [hunk 1]
```diff
@@ -262,6 +283,29 @@ pub mod pallet {
...
 
+    #[pallet::storage]
+    #[pallet::getter(fn dpr_price)]
+    pub(super) type DprPrice<T: Config> = StorageValue<_, BalanceOf<T>, OptionQuery>;
+
+    #[pallet::storage]
+    #[pallet::getter(fn price_diff_rate)]
+    pub(super) type PriceDiffRate<T: Config> = StorageValue<_, Percent, OptionQuery>;
+
+    /// tupule (BalanceOf<T>,BalanceOf<T>): first usdt amount, second
```
Comment (xcaptain) [inline pallets/credit/src/lib.rs:294]: second what?

File: pallets/credit/src/lib.rs [hunk 2]
```diff
@@ -1033,6 +1136,69 @@ pub mod pallet {
...
+                return (0u32.into(), weight);
+            }
+            let staking_usdt = staking_balance.unwrap().0;
+            let price = Self::dpr_price();
+            weight = weight.saturating_add(T::DbWeight::get().reads_writes(1, 0));
+            if price.is_none() {
+                return (0u32.into(), weight);
+            }
+            let dpr_amount =
+                staking_usdt / price.unwrap() * UniqueSaturatedFrom::unique_saturated_from(DPR);
```
Comment (xcaptain) [inline pallets/credit/src/lib.rs:1157]: Whether we should use `staking_usdt * price` here?

if `price` means one 1 USDT equals like `25_000_000_000_000_000` (0.025DPR), staking `x` USDT should got `x * price` DPR.

File: pallets/credit/src/lib.rs [hunk 3]
```diff
@@ -1033,6 +1136,69 @@ pub mod pallet {
...
+
+            let current_credit_level = credit_data.current_credit_level;
```
Comment (xcaptain) [inline pallets/credit/src/lib.rs:1159]: This variable `current_credit_level` is used only once, no need to create it, can use `credit_data.current_credit_level` directly

File: pallets/credit/src/lib.rs [hunk 4]
```diff
@@ -1033,6 +1136,69 @@ pub mod pallet {
...
+
+        fn calc_normal_daily_poc_reward(credit_data: &CreditData) -> (BalanceOf<T>, Weight) {
+            let mut weight = Weight::zero();
+            let initial_credit_level = credit_data.initial_credit_level;
+            let credit_setting =
+                Self::credit_settings(credit_data.campaign_id, initial_credit_level);
+            weight = weight.saturating_add(T::DbWeight::get().reads_writes(1, 0));
+
+            // poc reward
+            let current_credit_level = credit_data.current_credit_level;
```
Comment (xcaptain) [inline pallets/credit/src/lib.rs:1176]: Same like line 1159, no need to create a new variable.

File: pallets/staking/src/lib.rs [hunk 5]
```diff
@@ -1037,6 +1038,21 @@ pub mod pallet {
             Self::do_staking_delegate(who, dst_level)
         }
 
+        #[pallet::weight(T::WeightInfo::usdt_staking_delegate())]
+        #[transactional]
+        pub fn usdt_staking_delegate(
+            origin: OriginFor<T>,
+            account_id: T::AccountId,
+            usdt_amount: BalanceOf<T>,
```
Comment (xcaptain) [inline pallets/staking/src/lib.rs:1046]: `USDT` only has 6 decimals, see https://etherscan.io/token/0xdac17f958d2ee523a2206206994597c13d831ec7
how about using a `u32` or `usize` type.

File: pallets/credit/src/lib.rs (hunk 2 above)
Comment (jerry-yu) [reply pallets/credit/src/lib.rs:1157]: ??

File: pallets/credit/src/lib.rs (hunk 3 above)
Comment (jerry-yu) [reply pallets/credit/src/lib.rs:1159]: what't problem

File: pallets/credit/src/lib.rs (hunk 4 above)
Comment (jerry-yu) [reply pallets/credit/src/lib.rs:1176]: what't problem

File: pallets/staking/src/lib.rs (hunk 5 above)
Comment (jerry-yu) [reply pallets/staking/src/lib.rs:1046]: according to our smart contract

File: pallets/credit/src/lib.rs (hunk 2 above)
Comment (xcaptain) [reply pallets/credit/src/lib.rs:1157]: Sorry, my mistake, you are right.

File: pallets/staking/src/lib.rs (hunk 5 above)
Comment (xcaptain) [reply pallets/staking/src/lib.rs:1046]: not a big deal, keep BalanceOf<T> is ok for me.

**Review by xcaptain — APPROVED** [review]

**Pull request merged**